### ISSUE TITLE
Online Hard Example Mining

### ISSUE THREAD

**Issue by bhack** [tensorflow/tensorflow#17693]
Is there a plan for OHEM in `tf.estimator.train_and_evaluate` or in combo with dataset api?

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce

**Comment by jhseu** [comment]
Reassigning to @xiejw who wrote train_and_evaluate. I doubt there's a plan, but if you have any suggestions, we're open to them.

**Comment by bhack** [comment]
I think that generally there is a need of an intereplay between estimator train process and dataset api. Other than for OHEM this kind of interaction between could be useful for other tasks like identify labels with noise in a dataset etc.

**Comment by xiejw** [comment]
assign @ispirmustafa for planning.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @ispirmustafa: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by ispirmustafa** [comment]
Hi @bhack 
We don't have any plan re: OHEM. Could you please elaborate more about your proposal?
Thank you

**Comment by bhack** [comment]
Yes there are different approaches but just to make an example:
http://www.erogol.com/online-hard-example-mining-pytorch/

With high level api i.e. estimators, if we want to automate this process, we need a way to let estimators to interact with the dataset api and its iterators to handle hard cases mining.

There was also an old request on Keras: https://github.com/keras-team/keras/issues/6569

See also OHEM section in https://arxiv.org/pdf/1604.03540.pdf

**Comment by ispirmustafa** [comment]
Closing since there isn't any plan. Having said that @bhack if you have come up examples i'll be happy to review.
